resqpy/organize.py

```diff
@@ -955,69 +756,52 @@ def create_xml(self, organization_feature_root = None,
-      self.feature_root = None
+      self.domain = domain
+      self.genetic_boundary_feature = genetic_boundary_feature  # InterpretedFeature RESQML field, when not loading from xml
+      self.feature_root = None if self.genetic_boundary_feature is None else self.genetic_boundary_feature.root
+      if (not title) and self.genetic_boundary_feature is not None: title = self.genetic_boundary_feature.feature_name
       self.has_occurred_during = (None, None)
-      self.boundary_relation_list = None
-      self.sequence_stratigraphy_surface = None
-      self.extra_metadata = {}
-
-      if extract_from_xml and self.root_node is not None:
-         self.uuid = self.root_node.attrib['uuid']
-         self.domain = rqet.find_tag_text(self.root_node, 'Domain')
-         interp_feature_ref_node = rqet.find_tag(self.root_node, 'InterpretedFeature')
-         assert interp_feature_ref_node is not None
-         self.feature_root = self.model.referenced_node(interp_feature_ref_node)
-         if self.feature_root is not None:
-            self.genetic_boundary_feature = GeneticBoundaryFeature(self.model, kind = 'horizon',
-                                                root_node = self.feature_root,
-                                                feature_name = self.model.title_for_root(self.feature_root))
-         self.has_occurred_during = extract_has_occurred_during(self.root_node)
-         br_node_list = rqet.list_of_tag(self.root_node, 'BoundaryRelation')
-         if br_node_list is not None and len(br_node_list) > 0:
-            self.boundary_relation_list = []
-            for br_node in br_node_list:
-               self.boundary_relation_list.append(br_node.text)
-         self.sequence_stratigraphy_surface = rqet.find_tag_text(self.root_node, 'SequenceStratigraphySurface')
-         self.extra_metadata = rqet.load_metadata_from_xml(self.root_node)
-      else:
-         assert genetic_boundary_feature is not None
-         assert domain in ['depth', 'time', 'mixed'], 'unrecognised domain value for horizon interpretation'
-         if sequence_stratigraphy_surface is not None:
-            assert sequence_stratigraphy_surface in ['flooding', 'ravinement', 'maximum flooding', 'transgressive']
-            self.sequence_stratigraphy_surface = sequence_stratigraphy_surface
-         self.uuid = bu.new_uuid()
-         self.genetic_boundary_feature = genetic_boundary_feature
-         self.domain = domain
-         self.boundary_relation_list = boundary_relation_list
-         if self.boundary_relation_list is not None:
-            for boundary_relation in self.boundary_relation_list:
-               assert boundary_relation in ['conformable', 'unconformable below and above',
-                                            'unconformable above', 'unconformable below']
-
-      if self.uuid is None: self.uuid = bu.new_uuid()
+      self.boundary_relation_list = boundary_relation_list.copy()
```

Comment:
This will throw an exception when the default args are left: `None` cannot be copied

Reply:
Ah, that is a bug then

Reply:
I can fix this one.